# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../smalltalk/basePath.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- smalltalk-2.3.3/scripts/Finish.st_orig	2007-03-24 11:11:28.000000000 +0100
+++ smalltalk-2.3.3/scripts/Finish.st	2007-03-24 11:38:08.000000000 +0100
@@ -43,11 +43,12 @@
 
 "Remove DESTDIR and references to the build directory, from the paths
  stored in the image"
-| newImagePath newKernelBasePath |
+| newImagePath oldKernelBasePath newKernelBasePath |
+oldKernelBasePath := File pathFor: KernelFilePath.
 newKernelBasePath := Smalltalk arguments at: 1.
 newImagePath := Smalltalk arguments at: 2.
-KernelFilePath = newKernelBasePath
-    ifFalse: [ FileSegment relocateFrom: KernelFilePath to: newKernelBasePath ].
+oldKernelBasePath = newKernelBasePath ifFalse: [
+    FileSegment relocateFrom: oldKernelBasePath to: newKernelBasePath ].
 
 ImageFileName := 'gst.im'.
 ImageFilePath := newImagePath.
